src/main/java/nextstep/subway/section/domain/Sections.java

```diff
@@ -0,0 +1,119 @@
+package nextstep.subway.section.domain;
+
+import nextstep.subway.station.domain.Station;
+
+import javax.persistence.CascadeType;
+import javax.persistence.Embeddable;
+import javax.persistence.OneToMany;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+@Embeddable
+public class Sections {
+    public static final String HAS_UP_AND_DOWN_STATION_MSG = "상행역과 하행역이 이미 다른 구간에 동시에 등록되어 있습니다.";
+    public static final String HAS_NOT_UP_AND_DOWN_STATION_MSG = "새로운 구간에 상행역과 하행역 중 하나는 포함되어야 합니다.";
+
+    @OneToMany(mappedBy = "line", cascade = CascadeType.ALL, orphanRemoval = true)
+    private List<Section> sections = new ArrayList<>();
+
+    public List<Station> getStationsSorted() {
+        List<Station> stations = new ArrayList<>();
+
+        Station rootStation = getRootStation();
+        stations.add(rootStation);
+        Station nextStation = getNextStation(rootStation);
+
+        while(Objects.nonNull(nextStation)) {
+            stations.add(nextStation);
+            nextStation = getNextStation(nextStation);
+        }
+
+        return stations;
+    }
+
+    private Station getNextStation(Station downStation) {
+        Section nextSection = sections.stream()
+                .filter(section -> section.getEqualsUpStation(downStation))
+                .findFirst()
+                .orElse(null);
+
+        if(Objects.nonNull(nextSection))
+            return nextSection.getDownStation();
```

Comment:
```suggestion
        if (Objects.nonNull(nextSection)) {
            return nextSection.getDownStation();
        }
```

자바의 경우 한 라인도 괄호로 감싸주는 것이 관례입니다~